test/Microsoft.Azure.Devices.Edge.Test/AuthorizationPolicy.cs

```diff
@@ -134,34 +135,24 @@ public async Task AuthorizationPolicyUpdateTest()
                 },
                 token);
 
-            EdgeModule edgeHub2 = deployment2.Modules[ModuleName.EdgeHub];
-            await edgeHub2.WaitForReportedPropertyUpdatesAsync(
-                new
-                {
-                    properties = new
-                    {
-                        reported = new
-                        {
-                            lastDesiredStatus = new
-                            {
-                                code = 200,
-                                description = string.Empty
-                            }
-                        }
-                    }
-                },
-                token);
-
-            var leaf = await LeafDevice.CreateAsync(
-                deviceId2,
-                Protocol.Mqtt,
-                AuthenticationType.Sas,
-                Option.Some(this.runtime.DeviceId),
-                false,
-                CertificateAuthority.GetQuickstart(),
-                this.iotHub,
-                token,
-                Option.None<string>());
+            // There is no reliable way to signal when the policy
+            // is updated in $edgehub, so need to retry several times.
+            //
+            // DefaultProgressive => 55 sec max.
+            LeafDevice leaf = null;
+            await RetryPolicy.DefaultProgressive.ExecuteAsync(async () =>
```

Comment:
So now we basically just poll for a while?

Reply:
yep...